app/jobs/base.rb

```diff
@@ -352,6 +354,8 @@ def perform(*args)
         self.class.clear_cluster_concurrency_lock!
       end
 
+      Jobs.decrease_batch_count(batch_id) if batch_id
```

Comment:
What if a job never completes successfully and exhausts the maximum number of retries? Do we want to handle cases like this or do we just put a giant warning in the API that all errors in the jobs that are enqueued must be rescued in order for the "callback" job to be enqueued?

Reply:
A success or failure is treated the same way.

Reply:
What is our definition of failure here? Is it the job exhausting the number of retired and ending up in the dead queue or is it just the job failing once?